server/remote_cache/action_cache_server/action_cache_server.go

```diff
@@ -203,6 +229,18 @@ func (s *ActionCacheServer) GetActionResult(ctx context.Context, req *repb.GetAc
 	if err != nil {
 		return nil, err
 	}
+	if isRestricted(req.GetInstanceName()) {
```

Comment:
Can you extract this logic out to a function and use it in both GetActionResult and UpdateActionResult?

Reply:
Done!